prefect_azure/credentials.py

```diff
@@ -64,18 +67,20 @@ def inner(*args, **kwargs):
     return outer
 
 
-@dataclass
-class BlobStorageAzureCredentials:
+class BlobStorageAzureCredentials(Block):
     """
-    Dataclass used to manage Blob Storage authentication with Azure.
+    Block used to manage Blob Storage authentication with Azure.
     Azure authentication is handled via the `azure` module through
     a connection string.
 
     Args:
         connection_string: includes the authorization information required
     """
 
-    connection_string: str
+    _block_type_name = "Blob Storage Azure Credentials"
+    _logo_url = "https://github.com/PrefectHQ/orion/blob/main/docs/img/collections/azure.png?raw=true"  # noqa
+
```

Comment:
"Azure Blob Storage Credentials" reads a little more naturally, I think. Might make sense to update the class name too?
```suggestion
    _block_type_name = "Azure Blob Storage Credentials"
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
```